### PR TITLE
Bugfix/red 137 iframe resize

### PR DESCRIPTION
**Pull request by red-tt-bear**

### CODE REVIEW

File: src/Assignments/ProblemIframe.tsx [hunk 1]
```diff
@@ -139,39 +128,35 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
...
         setLoading(false);
     };
 
     return (
         <>
             { loading && <Spinner animation='border' role='status'><span className='sr-only'>Loading...</span></Spinner>}
             {error && <div>{error}</div>}
-            <iframe
+            <IframeResizer
+                // Using onInit instead of ref because:
```
Comment (PanopticaRising) [inline src/Assignments/ProblemIframe.tsx:143]: Nice finds.

File: src/Assignments/ProblemIframe.tsx [hunk 2]
```diff
@@ -139,39 +128,35 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
...
+                // ref never get's set and a warning saying to use `forwardRef` comes up in the console
+                // Using forwardRef does not give you access to the iframe, rather it gives you access to 3 or 4 methods and properties (like `sendMessage`)
+                onInit={(iframe: IFrameComponent) => {
+                    // TODO do we need to unset the iframeref? As of right now it should not be required since it is always present within the component
+                    // If using dom elements the useRef is "Read Only", however I want control!
+                    (iframeRef as any).current = iframe;
```
Comment (PanopticaRising) [inline src/Assignments/ProblemIframe.tsx:149]: I'd say, since we can hook-in via the event handlers, we don't need a ref for anything.

**Review by PanopticaRising — APPROVED** [review]

**Review by doombot-exe — APPROVED** [review]

File: src/Assignments/ProblemIframe.tsx (hunk 2 above)
Comment (doombot-exe) [reply src/Assignments/ProblemIframe.tsx:149]: upcoming changes for save and submit use the ref for grabbing the form element and GeoGebra-related scripts in the iframe

Comment (doombot-exe) [reply src/Assignments/ProblemIframe.tsx:149]: have not tested the modified save/submit with these changes though...

**Pull request merged**